### DISCUSSION TITLE
rsyslog tls support?

### DISCUSSION THREAD

**Discussion by the-infrequency** [wazuh/wazuh#13993] [Q&A] [answered]
Hey,

I'm currently in the process of setting up rsyslog so Wazuh can analyse logs from a few different machines on the network. Thing is, rsyslog sends data in clear text, and while the current network setup is secure it never hurts to have "defense in depth". 

Rsyslog [supports TLS encryption](https://www.rsyslog.com/files/temp/doc-indent/tutorials/tls.html), but cannot find any information if Wazuh is able to recieve logs in this way. Anyone got any pointers as to how to set this up in Wazuh, or if it is even possible?

I guess one alternative would be to setup macsec between the Wazuh server and the remote host, but would be nice to keep it simple.

**Comment by mdiego92** [comment, marked as answer]
Hello the-infrequency,

Could you please follow this [documentation](https://wazuh.com/blog/how-to-configure-rsyslog-client-to-send-events-to-wazuh/) on how to configure rsyslog to send events to Wazuh and this [documentation](https://www.thegeekdiary.com/how-to-configure-rsyslog-server-to-accept-logs-via-ssl-tls/) on how to configure rsyslog server to accept logs via TLS? Your `ossec.conf` file should be configured as shown [here](https://documentation.wazuh.com/current/user-manual/capabilities/log-data-collection/how-it-works.html?highlight=remote#remote-syslog) to receive those logs:

```
<ossec_config>
  <remote>
    <connection>syslog</connection>
    <port>513</port>
    <protocol>tcp</protocol>
    <allowed-ips>192.168.2.0/24</allowed-ips>
  </remote>
</ossec_config>
```

After the configuration is done, you’ll need to check if the logs are reaching the Wazuh Manager; for this, a TCP dump is needed:

`tcpdump -i any <protocol> -nn port 514 -AA
`

If you see packets incoming from the IP of the device sending the rsyslog, we need to check if the logs are being processed by the Wazuh Manager. To check this, you’ll have to enable the [logall_json](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html?highlight=logall#logall-json) option in the `ossec.conf` file 

```
<ossec_config>
<global>
...
<logall_json>yes</logall_json>
...
</global>
```

Then you should start seeing the logs in the `/var/ossec/logs/archives/archives.json` file. You can search for the device’s IP to see if logs are being processed as expected: `cat /var/ossec/logs/archives/archives.json | grep '"location":"<DEVICE-IP"'`

**Please remember disabling the `logall_json` option after the tests, as Wazuh stores all received events in the file archives.json even if they do not trip a rule, this is to avoid excessive disk storage consumption.**

In addition to this, you might also need [custom rules and decoders](https://documentation.wazuh.com/current/user-manual/ruleset/custom.html) after you’re able to get those logs, so they can be digested as you need.

Please let us know if you have any remaining questions.

Thank you very much.

**Reply by the-infrequency** [reply]
Thank you kindly. That seems to have done the trick. Now to get stuck into the custom rules and decoders.
